packages/dd-trace/src/runtime_metrics.js

```diff
@@ -7,11 +7,19 @@ const os = require('os')
 const { DogStatsDClient } = require('./dogstatsd')
 const log = require('./log')
 const Histogram = require('./histogram')
-const { performance } = require('perf_hooks')
+const { performance, PerformanceObserver } = require('perf_hooks')
 
+const { NODE_MAJOR, NODE_MINOR } = require('../../../version')
 const INTERVAL = 10 * 1000
 
+// Node >=16 has PerformanceObserver with `gc` type, but <16.7 had a critical bug.
+// See: https://github.com/nodejs/node/issues/39548
+const hasGCObserver = NODE_MAJOR >= 18 || (NODE_MAJOR === 16 && NODE_MINOR >= 7)
+const hasGCProfiler = NODE_MAJOR >= 20 || (NODE_MAJOR === 18 && NODE_MINOR >= 15)
```

Comment:
Shouldn't we test for major versions being `>= 17` and `>= 19`, respectively? Otherwise we disable these for people that might run Node versions 17 and 19 (I know it is unlikely, but not impossible.) 19.9.0 is still offered for download.

Reply:
We do not support odd versions of Node except when it's the current version, and any fixes that may have landed in 18 and even backported to 16 would be missing from 17 for example, so in my opinion it's safer to just leave them out.

Reply:
Understood, makes sense.